### DISCUSSION TITLE
is mask  input possible?

### DISCUSSION THREAD

**Discussion by AlexD81** [Jonseed/ComfyUI-Detail-Daemon#6] [Ideas]
Hi, is it possible to add strength map input, or just mask input for the sampler(s),      this would significantly improve this addon,  because, some parts of image may already have enough details,  and they become over-detailed, and mask would solve this.
thank you

**Comment by Jonseed** [comment]
You can already use Detail Daemon (and the other nodes) in inpainting workflows, and the detail will only apply to the masked areas. Just mask the area you want to add detail to in the Load Image node (right-click the node and select "Open in MaskEditor"). I've added an [inpainting example workflow](https://github.com/Jonseed/ComfyUI-Detail-Daemon/blob/main/example_workflows/Flux%20inpainting-Detail%20Daemon.json) that shows how to use Detail Daemon in inpainting. Note that how much detail is added depends both on the parameters in the Detail Daemon node and also your `denoise` factor set in the BasicScheduler.

![Screenshot 2024-11-02 110729](https://github.com/user-attachments/assets/6c843b05-155e-4318-95ab-9f05826626bf)

**Reply by AlexD81** [reply]
thanks for the tip,  it may be a workaround,   but for those who not using inpainting,  just full img2img generation,  a mask or 'ammount map' input still would be great addition

**Reply by blepping** [reply]
> a mask or 'ammount map' input still would be great addition

it's not really possible to just apply because the way we're doing this is calling the model with an adjusted sigma. pretty much the only thing you could is have a mask between the areas DD applies and where it doesn't and blend based on that _but_ that would require calling the model twice every time - once for the DD version, once for the normal one. in other words, it would halve your generation speed which probably isn't worth it.

**Reply by Jonseed** [reply]
> but for those who not using inpainting, just full img2img generation, a mask or 'ammount map' input still would be great addition

It seems that if you are using a mask, then you are doing _inpainting_, not full img2img generation. Any time a mask is involved, you are saying that you only want the system to change certain parts of the image, which is inpainting. I know you want to do a full img2img process, with details enhanced only in certain parts of the image, but that is getting into _inpainting_ when you are using a mask.

Another way of doing this would be to first do inpainting on the part of the image that you want to have more detail (with Detail Daemon), and then do another pass inverting the mask without Detail Daemon. It makes for a more complex workflow, but it is doable in ComfyUI.

**Reply by AlexD81** [reply]
thanks for explanation  that it's not possible within single pass.  Addon is still great, thank for your work, already improved quality of generations. In my projects, strength map of details is must have feature that is missing in comfy itself, I used to control level of details via differential diffusion or via another alternative,
SonarCompositeNoise, sadly it's difficult to adjust , @blepping could you take look pls to my question in [another](https://github.com/comfyanonymous/ComfyUI/discussions/4520#discussioncomment-10697408) thread, it works, but i have to change multiple parameters and on high resolution where generations are slow it becomes difficult

**Reply by blepping** [reply]
@AlexD81 
> via another alternative,
SonarCompositeNoise, sadly it's difficult to adjust , @blepping could you take look pls to my question in https://github.com/comfyanonymous/ComfyUI/discussions/4520#discussioncomment-10697408 thread

ahh, very sorry! i somehow missed replying to that one. i'll definitely never ignore someone intentionally, so thanks for the reminder. i'll write you a response there shortly.
